### ISSUE TITLE
Unknown device type for size iOS-5.8-in fastlane frameit

### ISSUE THREAD

**Issue by jonathanneilritchie** [fastlane/fastlane#10805]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When running 'fastlane frameit' in a directory with iPhone X screenshots (resolution 1125x2436) the device type cannot be found. 

##### Complete output when running fastlane, including the stack trace and command used

INFO [2017-11-06 13:08:24.48]: Framing screenshot './Simulator Screen Shot - iPhone X - 2017-11-06 at 12.35.35.png'
ERROR [2017-11-06 13:08:24.48]: Unknown device type for size iOS-5.8-in for path './Simulator Screen Shot - iPhone X - 2017-11-06 at 12.35.35.png'
ERROR [2017-11-06 13:08:24.49]: Unknown device type for size iOS-5.8-in for path './Simulator Screen Shot - iPhone X - 2017-11-06 at 12.35.35.png'
ERROR [2017-11-06 13:08:24.49]: Unknown device type for size iOS-5.8-in for path './Simulator Screen Shot - iPhone X - 2017-11-06 at 12.35.35.png'
DEBUG [2017-11-06 13:08:24.49]: Looking for Apple true Space Gray and found 0 template(s)
ERROR [2017-11-06 13:08:24.49]: Couldn't find template for screenshot type 'Apple true Space Gray'
ERROR [2017-11-06 13:08:24.49]: Please run `fastlane frameit download_frames` to download the latest frames

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.4.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.13.6 (Apple Git-96)           |
| Installation Source         | ~/.rvm/gems/ruby-2.4.0/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G1036)                  |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.0/lib                |
| OpenSSL Version             | OpenSSL 1.1.0e  16 Feb 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.1                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.63.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| bundler                   | 1.14.5       |
| faraday_middleware        | 0.11.0.1     |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| addressable               | 2.5.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| jwt                       | 1.5.6        |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| json                      | 2.1.0        |
| io-console                | 0.4.6        |
| excon                     | 0.57.0       |
| openssl                   | 2.0.3        |
| plist                     | 3.3.0        |
| faraday                   | 0.12.1       |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| terminal-table            | 1.8.0        |
| unicode-display_width     | 1.3.0        |
</details>

*generated on:* **2017-11-06**
</details>

**Comment by diegotrevisan** [comment]
Same for me.

**Comment by diegotrevisan** [comment]
Fixed for me in 2.64.0

**Comment by joshdholtz** [comment]
@jonathanneilritchie Does `2.64.0` fix this for you?

**Comment by banaslee** [comment]
I was having the same issue and ```2.64.0``` fixed it. However, now the ```show_complete_frame``` setting is being ignored with iPhone X.

**Comment by joshdholtz** [comment]
@banaslee I have not done anything with iPhone X and `frameit` yet but I will take a look at it and see what is going on 💪

**Comment by getaaron** [comment]
Possibly related, but iPhone X frames aren't working for me at all in 2.64.0. With this scanfile:

```ruby
devices([
  "iPhone X",
  "iPhone 8 Plus",
  "iPhone 8",
  "iPhone SE",
])

languages(["en-US"])

clear_previous_screenshots true
```

If I run `bundle exec fastlane run snapshot && bundle exec fastlane run frameit`, the screenshots are recorded correctly but only the 8/8+/SE devices get framed.

**Comment by MoOx** [comment]
I didn't have any issue in my case and was able to generate iPhone X framed screenshots. However, I am not able to continue because of this error (when I try to deliver)

```
[!] Unsupported screen size [1295, 2590] for path './fastlane/screenshots/en-US/iPhone X-01HomeScreen_framed.png'
```

**Comment by taquitos** [comment]
It looks like frameit is not resizing the images to fit the size requirements of iTunesConnect `frameit` repo:
https://github.com/fastlane/frameit-frames/blob/gh-pages/1509548869/Apple%20iPhone%20X%20Silver.png shows `1295x2590`, 

My understanding is that after getting screen shots, it places the frame around it, then it resizes the resulting image to be the correct size required by itunes connect. It's this last part I think that isn't happening.

**Comment by getaaron** [comment]
@jonathanneilritchie can you post the contents of your Framefile?

**Comment by jonathanneilritchie** [comment]
@getaaron I'm not using a Framefile at the moment. Just framing screenshots as needed from the terminal - works well for all other devices except the iPhone X.

**Comment by zocario** [comment]
I confirm that `show_complete_frame` is being ignored with iPhone X still have complete frame even if set to `false`

**Comment by taquitos** [comment]
Alright, so there are a few things here. 

- frameit isn't designed to work with App Store... it's meant for marketing materials, and websites, etc...
- frameit docs don't do a good job of pointing that out, in-fact, they are kind misleading (I'll fix that)
- `show_complete_frame: false` only works if you have a title and a background images

**Comment by getaaron** [comment]
I thought frameit was supposed to make App Store sized assets if you use a framefile. At least, that’s my understanding of the docs.

**Comment by zocario** [comment]
@taquitos I agree documentation is confusing: `frameit allows you to put a gorgeous device frame around your iOS and macOS screenshots just by running one simple command. Use frameit to prepare perfect screenshots for the App Store, your website, QA or emails.`. So my understanding was the same and I'm pretty sure many companies are using `frameit` to automate / send marketing screenshot directly to the AppStore. 

What about adding an optional parameter to respect AppStore screenshot size? It'd be really disappointing to lose possibility to generate screenshot for the AppStore with `frameit`.

**Comment by zocario** [comment]
And for `show_complete_frame` I have an example of the output with title and background:

![iphone x-issues copie_framed](https://user-images.githubusercontent.com/800617/32879451-53fd028a-caaa-11e7-8fbb-d20743e2c327.png)

My framefile:
```
{
  "default": {
       "title": {
	    "font": "./fonts/MuseoSans-700.otf",
      	    "color": "#ffffff"
        },
        "background": "./background.jpg",
        "padding": 100,
        "show_complete_frame": false
   },
   "data": [
	   {
		   "filter": "Issues",
                   "background": "./background 2.png"
	   }
   ]
}
```

**Comment by taquitos** [comment]
@zocario and @getaaron I totally agree, it would be amazing if we could support the App Store out of the box. The challenge is that each frame is actually larger than the screenshot. The frames also have different width-to-height ratio than the screenshots, so you can't just apply a simple % reduction after inserting the screenshot into the frame. You actually need to have a background that matches the correct app store required size, then you can reduce and compose the framed image into the background.

**Comment by michaelScoff** [comment]
@taquitos what'st the status of https://github.com/fastlane/fastlane/issues/10805#issuecomment-344842290. Can we find a workaround for iPhone X? 

@zocario did you find a workaround to properly frame screenshots?

**Comment by michaelScoff** [comment]
This is not being fixed, I would like to bring this up.

**Comment by bitomule** [comment]
Any update on this?

**Comment by inigo333** [comment]
Still an issue in fastlane 2.81.0
iPhone X screenshots is completely ignoring show_complete_frame
I'm setting show_complete_frame = false but still getting full frame.
Any updates?
Thanks

**Comment by RM335** [comment]
I faced the same problem and used ImageMagick CLI to resize the image to the required screenshot specifications. First I copied the images to a device specific folder and than run the command.

The iPhone 8 Plus:
`convert -gravity center "*.png" -resize 1242x2208 -extent 1242x2208 resized.png`

iPhone X:
`convert -gravity center "*.png" -resize 1125x2436 -extent 1125x2436 resized.png`

Eventually I ended up writing an alias for these commands. 
`alias resize-to-iphonex='convert -gravity center "*.png" -resize 1125x2436 -extent 1125x2436 resized.png'`
and
`alias resize-to-iphone8-plus='convert -gravity center "*.png" -resize 1242x2208 -extent 1242x2208 resized.png'`

**Comment by funnel20** [comment]
2 users have confirmed this original issue has been fixed in `2.64.0`, so I propose to close this issue:

- https://github.com/fastlane/fastlane/issues/10805#issuecomment-342327370
- https://github.com/fastlane/fastlane/issues/10805#issuecomment-342737291

Some users mention the issue that iPhone X ignores `show_complete_frame = false`, this can be followed in #11160.

**Comment by joshdholtz** [comment]
@funnel20 Thanks for the comment! I will close this issue since that new issue is opened up 😊
